### PR TITLE
Refactorings from cargo clippy etc.

### PR DESCRIPTION
**Pull request by topecongiro**
This PR

1. applies refactorings from `cargo clippy`.
2. refactors `write_snippet_inner()`. First split it into `process_comment()` and `process_missing_code()`, then simplify each functions.

### CODE REVIEW

File: src/codemap.rs [hunk 1]
```diff
@@ -94,7 +94,7 @@ impl LineRangeUtils for CodeMap {
         );
 
         LineRange {
-            file: lo.file.clone(),
+            file: Rc::clone(&lo.file),
```
Comment (nrc) [inline src/codemap.rs]: This seems like a bad change to me

File: src/bin/cargo-fmt.rs [hunk 2]
```diff
@@ -228,7 +228,7 @@ fn get_targets(strategy: &CargoFmtStrategy) -> Result<HashSet<Target>, io::Error
     if targets.is_empty() {
         Err(io::Error::new(
             io::ErrorKind::Other,
-            format!("Failed to find targets"),
+            String::from("Failed to find targets"),
```
Comment (nrc) [inline src/bin/cargo-fmt.rs]: I prefer `to_owned` to `String::from`

File: src/missed_spans.rs [hunk 3]
```diff
@@ -20,6 +20,25 @@ use shape::{Indent, Shape};
 use utils::{count_newlines, last_line_width, mk_sp};
 use visitor::FmtVisitor;
 
+struct SnippetStatus {
+    /// An offset to the current line from the beginnig of the original snippet.
+    line_start: usize,
+    /// A length of trailing whitespaces on the current line.
+    last_wspace: Option<usize>,
+    /// Teh current line number.
```
Comment (nrc) [inline src/missed_spans.rs]: typo: The

File: src/codemap.rs (hunk 1 above)
Comment (topecongiro) [reply src/codemap.rs]: The rational behind this is (taken from clippy's document):

> #### What it does
>Checks for usage of .clone() on a ref-counted pointer, (Rc, Arc, rc::Weak, or sync::Weak), and suggests calling Clone on the corresponding trait instead.
>#### Why is this bad
>: Calling ‘.clone()’ on an Rc, Arc, or Weak can obscure the fact that only the pointer is being cloned, not the underlying data.

File: src/missed_spans.rs (hunk 3 above)
Comment (topecongiro) [reply src/missed_spans.rs]: Thanks!

**Comment by topecongiro** [comment]
Thank you for the reviews, updated with the reviews reflected.

File: src/codemap.rs (hunk 1 above)
Comment (nrc) [reply src/codemap.rs]: I think Clippy is wrong here. I don't see why it is useful to selectively apply this rule, nor why there is a significant difference between cloning a smart pointer and some other data type (other data can be small too).

Comment (topecongiro) [reply src/codemap.rs]: Yeah just using a plain `.clone()` seems better. https://github.com/rust-lang-nursery/rustfmt/pull/2271/commits/516f15aba154e2cc66b999d0c9613479f4d53275 does not include this, so if there is no other problems, I think this PR can be merged.

**Pull request merged**